draft-ietf-wimse-workload-identity-bcp.md

```diff
@@ -63,45 +63,25 @@ informative:
-In addition, service account token volume projection allows an expiration time on these JWTs to be set, allowing automated rotation of these credentials.
-Finally, the private key for signing these tokens is managed by the "control plane", hence removing the manual effort of configuring the client_id and client_secret.
-
-However, there is currently no standardized way to manage these Service Account Tokens across container orchestrators.
-This leads to inconsistencies, and additional effort for developers as they need to support different client authentication mechanisms. In the worst case, this approach is ignored in favor of client_id and client_secret.
-
-This specification specifies the use of Service Account Tokens in container orchestration systems, which provides a secure and scalable way to create and manage these tokens, and ensures interoperability with existing OAuth-based authorization systems.
-
-When OAuth is used as part of the control plane entities, a Service Account Token is provisioned to the workload via the Agent. This interaction is shown in {{fig-arch}}.
-
-To distinguish the entities, we use the term "Control Plane" to refer to the OAuth 2.0 Authorization Server that is part
-of the cluster's control plane. Since there are also two access tokens in play, we use the term "Service Account Token" to refer to the token issued by the Control Plane and thereby distinguish it from the other access token issued to an OAuth 2.0 client running inside the workload by the second authorization server.
-
-It is important to note that the workload does not use the Service Account Token with resource servers directly but instead obtains access tokens from this second authorization server.
-To obtain these access tokens, the OAuth 2.0 client running in the workload uses the JWT client authentication grant, as defined in {{RFC7523}}, with the Service Account Token as input. The obtained access token may be a bearer token, or a proof-of-possession token.
+<!-- How Service Account Tokens can be/are used in combination with RFC 7523 to access OAuth2 protected resources -->
+Whilst the original purpose of the service account token was to authenticate access to the control plane API the industry has recognized its low maintainance and platform attestation capabilities and started using it as a JWT client assertion as specified in RFC 7523 {{RFC7523}}. The token is presented to a authorization server as a client assertion. The authorization servers validates the signature of the presented assertion via {{ OIDC metadata discovery }} or {{RFCXXXX}} and leverages the claims in the token to authenticate the client. Overall, the authorization server trusts the platform control plane with the issuance and delivery of these credentials. The authorization server responds with an Access Token the workload can use to access a OAuth2 protected resource on a resource server.
```

Comment:
You mention here that the authorization server validates signatures based on OIDC metadata discovery or RFCXXXX. Did you have something in mind here, which RFC to reference in addition?

Reply:
Sorry, I forgot to look up the RFC. It's RFC8414 which defines authorization server instance metadata under .well-known. It is very similar to OIDC but not compatible. It has differences, particulary how .well-known URI is constructed.